orca-core/src/main/java/com/netflix/spinnaker/orca/config/PreprocessorConfiguration.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.orca.config;
+
+import com.netflix.spinnaker.orca.extensionpoint.pipeline.ExecutionPreprocessor;
+import com.netflix.spinnaker.orca.extensionpoint.pipeline.ExecutionPreprocessorRegistry;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Configuration
+@EnableConfigurationProperties(DefaultApplicationConfigurationProperties.class)
+@ComponentScan("com.netflix.spinnaker.orca.preprocessors")
+public class PreprocessorConfiguration {
+
+  @Bean
+  ExecutionPreprocessorRegistry executionPreprocessorRegistry(Optional<List<ExecutionPreprocessor>> preprocessors) {
```

Comment:
This is currently different from how `PipelinePreprocessors` are registered, right? Is this the pattern we want to support going forward?

Reply:
Yes, correct.